src/kernels/gabor.jl

```diff
@@ -0,0 +1,74 @@
+import Base.getproperty
+
+"""
+    GaborKernel(; ell::Real=1.0, p::Real=1.0)
+
+Gabor kernel with length scale ell and period p. Given by
+```math
+    κ(x,y) =  h(x-z), h(t) = exp(-sum(t.^2./(ell.^2)))*cos(pi*sum(t./p))
+```
+
+"""
+struct GaborKernel{T<:Real, K<:Kernel} <: BaseKernel
+    κ::K
+    function GaborKernel(;ell=nothing, p=nothing)
+        k = _gabor(ell=ell, p=p)
+        if ell == nothing ell=1.0 end
+        if p == nothing p=1.0 end
+        new{Union{typeof(ell),typeof(p)}, typeof(k)}(k)
```

Comment:
It's not a good idea to define `T` is a union since that's an abstract type. One could use `promote_type` here, but I suggest removing `T` completely since it is not needed.
```suggestion
        k = _gabor(ell=ell, p=p)
        new{typeof(k)}(k)
```

Reply:
Of course, then `T` has to be removed in https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/pull/52/files#diff-d820aea21c04d68f343a510d8e894cf4R14.